server/server.go

```diff
@@ -2686,6 +3012,21 @@ func (s *StanServer) ioLoop(ready *sync.WaitGroup) {
 				remaining -= ioChanLen
 			}
 
+			replicateFutures := storeIOPendingMsgs(batch)
+
+			// If clustered, wait on the result of replication.
+			if s.isClustered() {
+				for i, future := range replicateFutures {
+					iopm = pendingMsgs[i]
+					if err := future.Error(); err != nil {
```

Comment:
Don't we need only a quorum to succeed, not all futures?  Or is that handled under the hood?

Reply:
That's happening under the hood in Raft. Each of these futures represents a separate Raft operation (in this case, replication of a batch of messages within a channel).